### PR TITLE
Wem 1055

### PR DESCRIPTION
**Pull request by shaniah868**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Review by RandyDeo — APPROVED** [review]

File: order_options.go [hunk 1]
```diff
@@ -1 +1,206 @@
...
+	Location string `json:"value"`
+	Type     string `json:"type"`
+}
+
+func (w *WhatsappOrderInfo) Validate() bool {
+	if w.Phone == "" {
+		w.ConvertedPhone = 0
+		return true
+	}
+	if len(w.Phone) <= 10 || len(w.Phone) >= 12 {
```
Comment (akeemphilbert) [inline order_options.go:57]: why not simply say len(w.Phone) != 11?

File: order_options.go [hunk 2]
```diff
@@ -1 +1,206 @@
...
+	}
+	i.PageName = name
+	return true
+}
+
+func (e *EmailOrderInfo) Validate() bool {
+	if e.Email == "" {
+		return true
+	}
+	_, isValid := IsValidUrl(e.Email, e.Type)
```
Comment (akeemphilbert) [inline order_options.go:90]: Is this checking for a valid email?

File: order_options.go [hunk 3]
```diff
@@ -1 +1,206 @@
...
+	return isValid
+
+}
+
+func (p *PhoneOrderInfo) Validate() bool {
+	if p.Phone == "" {
+		p.ConvertedPhone = 0
+		return true
+	}
+	if len(p.Phone) <= 10 || len(p.Phone) >= 12 {
```
Comment (akeemphilbert) [inline order_options.go:100]: there is a simpler way to represent this

File: order_options.go [hunk 4]
```diff
@@ -1 +1,206 @@
...
+	convertedPhone, err := strconv.Atoi(p.Phone)
+	if err != nil {
+		return false
+	}
+	p.ConvertedPhone = convertedPhone
+	return true
+}
+
+func (p *ProductUrlOrderInfo) Validate() bool {
+	return p.Url != ""
```
Comment (akeemphilbert) [inline order_options.go:112]: just return true here because this isn't define yet

**Review by akeemphilbert — CHANGES_REQUESTED** [review]

File: order_options.go (hunk 2 above)
Comment (shaniah868) [reply order_options.go:90]: Yes it is, I made if statements in IsValidUrl to validate urls and the emails

**Review by akeemphilbert — APPROVED** [review]

**Pull request merged**